cli/src/main/java/org/jboss/as/cli/CommandLineMain.java

```diff
@@ -151,4 +158,29 @@ private static Properties createLogManagerConfig(final String level) {
         properties.setProperty("formatter.PATTERN.properties", "pattern");
         return properties;
     }
+
+    /**
+     * Creates a file marker under $JBOSS_HOME/bin directory. This is used by the shutdown operation to understand from where we have
+     * launched CLI instance when we are performing an update of a remote installation.
+     */
+    private static void createClientMarker() {
+        try {
+            final String jbossHome = WildFlySecurityManager.getEnvPropertyPrivileged("JBOSS_HOME", null);
+            if (jbossHome != null) {
+                final Path cliMarkerPath = Paths.get(jbossHome).resolve("bin").resolve(Util.CLI_MARKER);
+                Files.deleteIfExists(cliMarkerPath);
```

Comment:
The RFE contemplates this situation where we have stated that is the customer the one that has to ensure no other processes are launched when they are updating the server.

In this case, the marker value returned by the customer won't match the current CLI instance, so it will assume it is not a local instance and won't be automatically closed.

The user will end up in the same situation as the current Patch tool, where an update/patch can update the underlying jboss-modules.jar. Since the client CLI session is not automatically closed, the user will be using an obsolete CLI process launched from a patched jboss-modules.jar. Probably he will get errors using the CLI and he will be forced to close it and open it again.